validations/validateNewCard.ts

```diff
@@ -0,0 +1,35 @@
+import { CARD_MAX_LENGTH, NEW_CARD_ERROR_MESSAGE } from '../constants';
+import { CardCompany, CardExpiredDate, CardNumber, CardPassword } from '../types/card';
+
+export const validateNewCard = (newCard: {
+  number: CardNumber;
+  cvc: string;
+  password: CardPassword;
+  expiredDate: CardExpiredDate;
+  cardCompany: CardCompany;
+}): void => {
+  const { number, expiredDate, cvc, password, cardCompany } = newCard;
+
+  const isValidCardNumberLength = Object.values(number).every(
+    (quarterCardNumber) => quarterCardNumber.length === 4,
+  );
+
+  const month = Number(expiredDate.month);
+  const year = Number(expiredDate.year);
+
+  const isValidExpiredMonth = month >= 1 && month <= 12;
+  const isValidExpiredYear = year >= 10 && year <= 50;
+
+  const isValidCvc = cvc.length === CARD_MAX_LENGTH.CVC;
+
+  const isValidPassword = Object.values(password).every((onePassword) => onePassword.length === 1);
+
+  const hasCompany = cardCompany.length > 0;
+
```

Comment:
각각의 validation 을 함수로 작성해서 테스트하는 건 어떨까요??

Reply:
```suggestion
const validateLength = len => val => val.length === len;
const validateMaxLength = len => val => val.length <= len;
```
요런 식으로 작성하면 각각의 함수를 단위테스트 짜기 용이 할것 같습니다!

Reply:
오 그런 이점이 있을 수 있겠네요!! 더 좋을 것 같아요!! 감사합니다!